src/Core/Util/String/RandomString.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Util\String;
+
+class RandomString
+{
+    public static function generate(int $length = 32): string
+    {
+        return substr(bin2hex(random_bytes($length)), 0, $length);
```

Comment:
Not blocking, but I think you could get a 14-long randomg string simply by calling `uniqid` 😄

Reply:
we talk here about crypto and security and 14-long random string like those generated by `uniqid` are not enough. You can have a look on "Client Secret" section here https://www.oauth.com/oauth2-servers/client-registration/client-id-secret/

Reply:
No there is no danger about using `uniqid()` to generate the API secret 😉 here's why:

Functions that do not generate cryptographically secure values can theoretically be guessable. However the guess does not come from nowhere 😄 .

If you use `uniqid()` to generate 1,000,000  values and the attacker can look at the 999,999 values, then he might be able to guess the 1,000,000th value. Attacker can use his knowledge of past generated values to guess the next one.

But I think we can agree no attacker will be able to find 999,999 secrets generated by our API, all coming from the same shop 😄 

On the opposite, a shop should not use `uniqid()` to generate things that can be quickly generated and seen by the attacker, like CSRF tokens or cookie tokens.

So cryptographically functions are needed for generation for things like CSRF tokens, cookie tokens, unique order IDs... things that could easily number in the hundred, thousand and more.

Reply:
I agree uniqid could be enough, but generating a longer secret is not that complex so why not do it Besides it's not the first time we encountered a use case where we needed a random string like this so having a handy tool to do it makes sense IMHO